test/Table_Tests/src/IO/Excel_Spec.enso

```diff
@@ -73,6 +73,41 @@ spec_fmt suite_builder header file read_method sheet_count=5 =
             t_3.at 'B' . to_vector . should_equal [Nothing, 30]
             t_3.at 'C' . to_vector . should_equal [43.2, 54]
 
+        group_builder.specify "should let you read all sheets from xlsx" <|
+            wb = read_method file
+
+            ## Read all the sheets as single rows.
+            single_table = wb.read_many
+            single_table.row_count . should_equal 25
+            single_table.column_names . should_equal ["SheetName", "Name", "Quantity", "Price", "A", "B", "C", "D", "E", "Student Name", "Enrolment Date", "Item", "Price 1"]
+            warnings = Warning.get_all single_table . map _.value
+            warnings.length . should_equal 2
+            warnings.should_contain Empty_Sheet.Error
+
+            ## Read all the sheets as tables.
+            table_of_tables = wb.read_many return=..Table_Of_Tables
+            table_of_tables.row_count . should_equal 5
+            table_of_tables.column_names . should_equal ["SheetName", "Table"]
+            table_of_tables.at "SheetName" . to_vector . should_equal ["Simple", "Strange Dimensions", "Dates", "Empty", "Duplicate Columns"]
+            Warning.get_all table_of_tables . map _.value . length . should_equal 2
+
+        group_builder.specify "should let you read some sheets from xlsx" <|
+            wb = read_method file
+            single_table = wb.read_many ["Simple", "Dates"]
+            single_table.row_count . should_equal 12
+            single_table.column_names . should_equal ["SheetName", "Name", "Quantity", "Price", "Student Name", "Enrolment Date"]
+            single_table.has_warnings.should_be_false
+
+        group_builder.specify "should let you read some sheets with a bad name from xlsx" <|
+            wb = read_method file
+            single_table = wb.read_many ["Simple", "Dates", "Not A Sheet"]
+            single_table.row_count . should_equal 12
+            single_table.column_names . should_equal ["SheetName", "Name", "Quantity", "Price", "Student Name", "Enrolment Date"]
+            single_table.has_warnings.should_be_true
+            warnings = Warning.get_all single_table . map _.value
+            warnings.length . should_equal 1
+            warnings.first.should_be_a Invalid_Location.Error
```

Comment:
Ideally I'd check the payload of the warning to ensure it is actually the "Not A Sheet". I don't expect that to fail but it'd make it more complete.

I just saw that `Problems.expect_warning` has a typo in its signature (it returns `Any`, not `Nothing`). It actually returns the warning if found, so it can be used for further inspection. 

Here we can probably prefer `Problems.expect_only_warning` that works as above but will additionally ensure no other unexpected warnings appear:
```suggestion
            warning = Problems.expect_only_warning Invalid_Location single_table
            warning.location . should_equal "Not A Sheet"
            warning.to_display_text . should_contain "Unknown sheet or range name or invalid address: Not A Sheet"
```